game-core/src/main/java/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -1268,6 +1309,16 @@ public Component getListCellRendererComponent(final JList<? extends Unit> list,
     return selection;
   }
 
+  private JOptionPane getOptionPane(final JComponent parent) {
+    JOptionPane pane = null;
```

Comment:
A ternary IMO would clean this up quite a bit:
```
return !(parent instanceof JOptionPane)
     ? getOptionPane((JComponent) parent.getParent())
     : (JOptionPane) parent;
```

If going with the old pattern, not setting the variable to `null` and then marking it `final` to ensure (compiler-enforced) that ti would get set in the if/else block. It's a signal to reader that the variable would be set. When I see a variable initialized to null like this and returned shortly later, it's a signal to me that the variable may not get set at all